test/libyul/yulOptimizerTests/commonSubexpressionEliminator/for_reassign_continue.sol

```diff
@@ -0,0 +1,22 @@
+{
+    let x := 0
+    for {} sload(0) { let t := 0 } {
+        x := 0
+        if calldataload(0) { x := 2 continue }
+    }
+}
+// ----
+// step: commonSubexpressionEliminator
+//
+// {
+//     let x := 0
+//     for { } sload(0) { let t := 0 }
```

Comment:
The optimizer currently replaces this by `let t := x`, which is wrong - `x` was reassigned in line 18.

To fix this, we need to store the state at each continue and break statement and take this properly into account like we do in the unused store base.